clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -713,13 +748,24 @@ private boolean appendsInProgress() {
      */
     public void awaitFlushCompletion() throws InterruptedException {
         try {
-            for (ProducerBatch batch : this.incomplete.copyAll())
+            for (ProducerBatch batch : this.incomplete.copyAll()) {
                 batch.produceFuture.await();
+                waitForPossibleSplittedBatches(batch);
```

Comment:
It seems like it might be cleaner if we move all the waitForPossibleSplittedBatches logic into a `ProducerBatch.await` method that performs the await on its produceFuture as well as on the child batches.

Reply:
Good idea,  I will change it accordingly.

Reply:
Done